jumpscale/sals/zos/container_stats.py

```diff
@@ -0,0 +1,99 @@
+from urllib.parse import urlparse
+import requests
+import json
+import redis
+import sys
+
+class ContainerStatsMonitor:
+    def __init__(self):
+        self.host = None
+        self.port = None
+        self.channel = None
+
+        self.redis = None
+        self.pubsub = None
+        self.previous = None
+
+    def endpoint(self, endpoint):
+        params = urlparse(endpoint)
+
+        self.host = params.netloc.split(':')[0]
+        self.port = params.netloc.split(':')[1]
+        self.channel = params.path[1:]
+
+        return self.connect()
+
+    #
+    # Reservation Parser
+    #
+    def reservation(self, url):
+        reservation = requests.get(url).json()
```

Comment:
should raise for status

Reply:
What do you mean ?

Reply:
the response object returned from `requests.get` have a `raise_for_status` method, which would raise an exception in case of non-successful requests (4xx or 5xx status code).

So, I think this can be like:

```python
resp = requests.get(url)
resp.raise_for_status()
reservations = resp.json()
```